controllers/workspace/predicates.go

```diff
@@ -0,0 +1,41 @@
+//
+// Copyright (c) 2019-2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package controllers
+
+import (
+	"github.com/devfile/api/pkg/apis/workspaces/v1alpha2"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+var predicates = predicate.Funcs{
```

Comment:
to understand why these predicates exist, it's needed to read the whole updateFunc.
I can't propose a really good solution on how to make it really transparent but maybe you have.
At least add a few lines of comment that we filter out all updates if a workspace is failed, unless routing is updated.

Reply:
Added a short comment c18edb9